Sources/GRPCCore/GRPCServer.swift

```diff
@@ -172,15 +190,39 @@ public final class GRPCServer: Sendable {
   ///       are called. The first interceptor added will be the first interceptor to intercept each
   ///       request. The last interceptor added will be the final interceptor to intercept each
   ///       request before calling the appropriate handler.
-  public init(
+  public convenience init(
     transport: any ServerTransport,
     router: RPCRouter,
     interceptors: [any ServerInterceptor] = []
   ) {
+    self.init(
+      transport: transport,
+      router: router,
+      interceptorPipeline: interceptors.map { .applyToAllServices($0) }
+    )
+  }
+
+  /// Creates a new server with no resources.
+  ///
+  /// - Parameters:
+  ///   - transport: The transport the server should listen on.
+  ///   - router: A ``RPCRouter`` used by the server to route accepted streams to method handlers.
+  ///   - interceptors: A collection of interceptors providing cross-cutting functionality to each
+  ///       accepted RPC. The order in which interceptors are added reflects the order in which they
+  ///       are called. The first interceptor added will be the first interceptor to intercept each
+  ///       request. The last interceptor added will be the final interceptor to intercept each
+  ///       request before calling the appropriate handler.
+  public init(
+    transport: any ServerTransport,
+    router: RPCRouter,
+    interceptorPipeline: [ServerInterceptorOperation]
```

Comment:
Previously the interceptors were passed to the router when `handle` is called. This meant that at this point in the code – i.e. when the router is passed to the server – that the router was fully configured and the caller was responsible for configuring it.

In the _other_ API, where the caller passes services and interceptors to the server, the server was responsible for configuring the router.

With the router now accepting interceptors the lines are blurred: it's not clear who is responsible for configuring the router. In this case it's partly the caller and partly the server. I think this distinction should be clearer.

I see a few options:
1. Remove the ability to create the server with a router _and_ interceptors and make the caller responsible for configuring the router entirely. (If the API where they pass it into the server.)
2. Stop storing the interceptors on the router and continue storing them on the server. This would require the server to lookup the interceptors for each RPC and pass them into the router (which is similar to the existing setup).

Reply:
Good point. I honestly think that 1 is the right approach here, since, in the init where the router is passed, we are already assuming it's set up properly (with e.g. each service's methods registered).

Reply:
Great, I'm happy with (1)